ktor-client/ktor-client-core/common/src/io/ktor/client/utils/ClientEvents.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.utils
+
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.events.*
+import kotlin.native.concurrent.*
+
+/**
+ * Happens after creating of the new request.
+ */
+@SharedImmutable
+public val HttpRequestCreated: EventDefinition<HttpRequestBuilder> = EventDefinition()
+
+/**
+ * Happens after execution of all interceptor before sending the request.
```

Comment:
Happens -> Occurs before sending the request, and after execution of all interceptors.